nmmo/task/predicate_api.py

```diff
@@ -54,12 +53,12 @@ def __call__(self, gs: GameState) -> float:
     for group in self._groups:
       group.update(gs)
     # Calculate score
-    cache = gs.cache_result
-    if self.name in cache:
-      progress = cache[self.name]
+    # cache = gs.cache_result
+    if self.name in gs.cache_result:
+      progress = gs.cache_result[self.name]
     else:
       progress = max(min(self._evaluate(gs)*1.0,1.0),0.0)
-      cache[self.name] = progress
+      gs.cache_result[self.name] = progress
```

Comment:
We have a test to check whether the results are cached and the tests pass, so I believe the results are cached. 

https://github.com/CarperAI/nmmo-environment/blob/task-rev/tests/task/test_predicates.py#L63

However, the new code is more neat. Thanks!